### PR TITLE
Anonymous signals [B-04332]

### PR DESCRIPTION
**Pull request by timotree3**
Implements anonymous signals and adds a test for them. The logic for anonymous signals pays attention to the cell ID given by appInfo, so I had to update the tests to use proper cell IDs.

Companion chaperone PR: https://github.com/Holo-Host/chaperone/pull/203 (Will need to make a chaperone release that includes that work before this can run CI)

### CODE REVIEW

**Comment by timotree3** [comment]
Tests are passing locally using the linked version of chaperone

File: src/index.ts [hunk 1]
```diff
@@ -237,6 +237,23 @@ class Envoy {
       const hha_hash = url.searchParams.get('hha_hash');
       log.normal("%s (%s) connection for HHA ID: %s", anonymous ? "Anonymous" : "Agent", agent_id, hha_hash);
 
+      // Signal is a message initiated in conductor which is sent to UI. In case to be able to route signals
```
Comment (robbiecarlton) [inline src/index.ts]:
```suggestion
      // Signal is a message initiated in conductor which is sent to UI. To be able to route signals
```

**Review by robbiecarlton — APPROVED** [review]

**Comment by robbiecarlton** [comment]
LGTM, once CI is passing

File: src/index.ts (hunk 1 above)
Comment (timotree3) [reply src/index.ts]: :P I didn't write that comment but that is an improvement.

**Pull request merged**